src/Data/Connection/PostObjectConnectionResolver.php

```diff
@@ -0,0 +1,380 @@
+
+		$arg_mapping = [
+			'authorName'    => 'author_name',
+			'authorIn'      => 'author__in',
+			'authorNotIn'   => 'author__not_in',
+			'categoryId'    => 'cat',
+			'categoryName'  => 'category_name',
+			'categoryIn'    => 'category__in',
+			'categoryNotIn' => 'category__not_in',
+			'tagId'         => 'tag_id',
+			'tagIds'        => 'tag__and',
+			'tagIn'         => 'tag__in',
+			'tagNotIn'      => 'tag__not_in',
+			'tagSlugAnd'    => 'tag_slug__and',
+			'tagSlugIn'     => 'tag_slug__in',
+			'search'        => 's',
+			'id'            => 'p',
+			'parent'        => 'post_parent',
+			'parentIn'      => 'post_parent__in',
+			'parentNotIn'   => 'post_parent__not_in',
+			'in'            => 'post__in',
+			'notIn'         => 'post__not_in',
+			'nameIn'        => 'post_name__in',
+			'hasPassword'   => 'has_password',
+			'password'      => 'post_password',
+			'status'        => 'post_status',
+			'stati'         => 'post_status',
+			'dateQuery'     => 'date_query',
+		];
+
+		/**
+		 * Map and sanitize the input args to the WP_Query compatible args
+		 */
+		$query_args = Types::map_input( $where_args, $arg_mapping );
+
+		if ( ! empty( $query_args['post_status'] ) ) {
+			$query_args['post_status'] = $this->sanitize_post_stati( $query_args['post_status'] );
+		}
+
+		/**
+		 * Filter the input fields
+		 * This allows plugins/themes to hook in and alter what $args should be allowed to be passed
+		 * from a GraphQL Query to the WP_Query
+		 *
+		 * @param array       $query_args The mapped query arguments
+		 * @param array       $args       Query "where" args
+		 * @param mixed       $source     The query results for a query calling this
+		 * @param array       $all_args   All of the arguments for the query (not just the "where" args)
+		 * @param AppContext  $context    The AppContext object
+		 * @param ResolveInfo $info       The ResolveInfo object
+		 * @param string      $post_type  The post type for the query
+		 *
+		 * @since 0.0.5
+		 * @return array
+		 */
+		$query_args = apply_filters( 'graphql_map_input_fields_to_wp_query', $query_args, $where_args, $this->source, $this->args, $this->context, $this->info, $this->post_type );
+
+		/**
+		 * Return the Query Args
+		 */
+		return ! empty( $query_args ) && is_array( $query_args ) ? $query_args : [];
+
+	}
+
+	/**
+	 * Limit the status of posts a user can query.
+	 *
+	 * By default, published posts are public, and other statuses require permission to access.
+	 *
+	 * This strips the status from the query_args if the user doesn't have permission to query for
+	 * posts of that status.
+	 *
+	 * @param $stati
+	 *
+	 * @return array|null
+	 */
+	public function sanitize_post_stati( $stati ) {
+		if ( empty( $stati ) ) {
+			$stati = [ 'publish' ];
+		}
+		$statuses = wp_parse_slug_list( $stati );
+		$post_type_obj = get_post_type_object( $this->post_type );
+		$allowed_statuses = array_filter( array_map(function( $status ) use ( $post_type_obj ) {
```

Comment:
missing space between `array_map(` and `function()`

BTW, this is the most javascript PHP code I've ever seen.

Reply:
I get so bummed when my es6 doesn't execute in PHP!! 😬 

Functional programming ftw!